querydsl-core/src/main/java/com/querydsl/core/support/SerializerBase.java

```diff
@@ -73,11 +73,35 @@ protected String getConstantPrefix() {
         return constantPrefix;
     }
 
-    public Map<Object,String> getConstantToLabel() {
-        if (constantToLabel == null) {
-            constantToLabel = new HashMap<Object,String>(4);
+    @Deprecated
+    public Map<Object, String> getConstantToLabel() {
+        return getConstantToAllLabels();
+    }
+
+    public Map<Object, String> getConstantToNamedLabel() {
+        if (constantToNamedLabel == null) {
+            constantToNamedLabel = new HashMap<Object, String>(4);
+        }
+        return constantToNamedLabel;
+    }
+
+    public Map<Object, Integer> getConstantToNumberedLabel() {
+        if (constantToNumberedLabel == null) {
+            constantToNumberedLabel = new HashMap<Object, Integer>(4);
         }
-        return constantToLabel;
+        return constantToNumberedLabel;
+    }
+
+    public Map<Object, String> getConstantToAllLabels() {
+        Map<Object, String> named = getConstantToNamedLabel();
+        Map<Object, Integer> numbered = getConstantToNumberedLabel();
+
+        Map<Object, String> combined = new HashMap<Object, String>(named);
+        for (Map.Entry<Object, Integer> entry : numbered.entrySet()) {
+            combined.put(entry.getValue(), entry.getValue().toString());
```

Comment:
It should be:
`combined.put(entry.getKey(), entry.getValue().toString());`

With the current code I get:
`Parameter value [1] did not match expected type [java.lang.String (n/a)]`

With the fix it works fine.